### DISCUSSION TITLE
Adding multiple comments to multiple folders (not files, folders)

### DISCUSSION THREAD

**Discussion by clausneergaard** [RhetTbull/osxmetadata#89] [Q&A]
Hello all -

I am fairly new to this Python-game, and I've been trying to work out how this stuff works. 

Though I get how to add comments to folders, one-by-one, I am fairly certain that it can be done with multiple folders, nested folders, etc. So, if I have a folder structure that goes something like this (I've managed to write code that takes this txt-file and creates all the folders, accordingly);

```
Annotations
Annotations/Fire
Annotations/Tags
Boundary Conditions
Cable Tray
```

I also have - and this is where I can't get it to work - another file containing all the comments that I want to add to each folder. They're all in the same order of appearance, and looks something like this;

```
"The Annotations-folder contains all Families in relation to annotation of objects."
"The Fire-folder contains all Families in relation to fire."
"The Tags-folder contains all Families in relation to Tags."
"The Boundary Conditions-folder contains all Families in relation to Boundary Conditions."
"The Cable Tray-folder contains all Families in relation to Cable Trays."
```

Here is the code, that I know works, but this only does it to one folder;

```
from osxmetadata import *

path = "/Users/clausneergaard/Library/CloudStorage/Dropbox/Projekter/21-01 Revit families, dokumentation/Folders (empty)/Folders/"
md = OSXMetaData(path)
md.kMDItemFinderComment = "This is a comment added via Python for MacOS."
```

As mentioned above, I would love for it to be able to take each created folder, and add a comment from a txt-file, that of course matches the folder structure.

Any help is appreciated. 
Thank you.

**Comment by RhetTbull** [comment]
Hi. I think what you want is [os.walk](https://docs.python.org/3.9/library/os.html?highlight=os.walk#os.walk) which will allow you walk a directory tree and handle each directory in the tree. If your comments and folder names are in a text file (maybe consider using [CSV](https://docs.python.org/3.9/library/csv.html?highlight=csv)) you could then find the corresponding match for each folder and apply the comment.  `osxmetadata` itself will only operate on a single folder or single file.

**Comment by clausneergaard** [comment]
Ah ... yeah, OK. I see, didn't know that osxmetadata only would work on a single folder og file.
Thank you - I'll give the `os.walk` a go, and I'll be back if I can't work it out with your inputs. 

Thanks!
